aptos-move/framework/aptos-stdlib/sources/cryptography/ed25519.move

```diff
@@ -166,6 +170,44 @@ module aptos_std::ed25519 {
         std::hash::sha3_256(pk_bytes)
     }
 
+    #[test_only]
+    public fun generate_keys(): (SecretKey, ValidatedPublicKey) {
+        let (sk_bytes, pk_bytes) = generate_keys_internal();
+        let sk = SecretKey {
+            bytes: sk_bytes
+        };
+        let pk = ValidatedPublicKey {
+            bytes: pk_bytes
+        };
+        (sk,pk)
+    }
+
+    #[test_only]
+    public fun sign_arbitrary_bytes(sk: &SecretKey, msg: vector<u8>): Signature {
+        Signature {
+            bytes: sign_internal(sk.bytes, msg)
+        }
+    }
+
+    #[test_only]
+    public fun sign_struct<T:drop>(sk: &SecretKey, data: T): Signature {
```

Comment:
Hm... This is interesting. We're taking this by value. I guess we are consistent with the verification API, so it's good.